qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewPresenter.cpp

```diff
@@ -294,7 +321,19 @@ void PreviewPresenter::runSumBanks() {
   // Ensure the angle is up to date so that we can check for matching experiment settings lookup rows
   m_model->setTheta(m_view->getAngle());
 
-  if (!m_model->getSelectedBanks().has_value() && !m_mainPresenter->hasROIDetectorIDsForPreviewRow()) {
+  auto const expSettingsDetectorROI = m_mainPresenter->getMatchingROIDetectorIDsForPreviewRow();
+  if (m_plotExistingROIs) {
+    auto const selectedDetectorsOnPlot = m_dockedWidgets->getSelectedDetectors();
```

Comment:
Can these be const refs?
```suggestion
  auto const &expSettingsDetectorROI = m_mainPresenter->getMatchingROIDetectorIDsForPreviewRow();
  if (m_plotExistingROIs) {
    auto const &selectedDetectorsOnPlot = m_dockedWidgets->getSelectedDetectors();
```

Reply:
Sorry, similar question on this one for `getSelectedDetectors`, as that method looks like it creates a vector, populates it and then returns it. So does something happen in the background that means that doesn't go out of scope once the method ends?